### ISSUE TITLE
Some child services didn't restarted with command `reload-or-restart` the parent target, if one of the services has ExecReload and ReloadPropagatedFrom options defined

### ISSUE THREAD

**Issue by vifreefly** [systemd/systemd#10638]
**systemd version the issue has been seen with**
```
$ systemctl --version

systemd 239 (same behavior on 237)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid
```

**Used distribution**
 > Ubuntu 18.10 (same behavior on 18.04)

**Expected behaviour you didn't see**
 If I want to reload/restart the main target (`$ sudo systemctl reload-or-restart main.target`) that should mean that all child services should be reloaded/restarted as well (if option ExecReload provided for the child service then reload, if not then just restart).

**Unexpected behaviour you saw**
 When I try to `reload-or-restart main.target` with units below, reloaded only service with `ExecReload` option. Another service (without ExecReload) just ignored instead of restarting.

**Steps to reproduce the problem**
Create 3 units (1 target and 2 child services, where one of them has ExecReload option and another don't):

main.target:
```
[Unit]
Description=Main target
Wants=with_reload.service without_reload.service

[Install]
WantedBy=multi-user.target
```

with_reload.service:
```
[Unit]
Requires=network.target
PartOf=main.target
ReloadPropagatedFrom=main.target

[Service]
Type=simple
User=deploy
WorkingDirectory=/home/deploy

ExecStart=<start command>
ExecReload=<reload command>

Restart=always
RestartSec=1
TimeoutStopSec=15
KillMode=mixed
StandardInput=null
```

without_reload.service:
```
[Unit]
Requires=network.target
PartOf=main.target

[Service]
Type=simple
User=deploy
WorkingDirectory=/home/deploy

ExecStart=<start command>

Restart=always
RestartSec=1
TimeoutStopSec=15
KillMode=mixed
StandardInput=null
```

**Comment by bl33pbl0p** [comment]
There's no hierarchy except dependency relationships here.

PartOf= is one way (propagate restart events in the listed unit to me). If you want the restart to be propagated from with_reload.service and without_reload.service to main.target, you need to list PartOf= in main.target.

In the current setup, reload or restart will flow down to with_reload/without_reload, not the other way around.

PS: The reverse of this property may serve as a useful addition (ConsistsOf= is automatic property right now) and avoid duplication in all units.

**Comment by vifreefly** [comment]
@arvidjaar Thanks for the answer! 

Can you please tell what is it, `JOB_RELOAD_OR_RESTART` ? How I can use it to make work correctly example above?

**Comment by arvidjaar** [comment]
> Can you please tell what is it, `JOB_RELOAD_OR_RESTART` ?

It does not yet exist, needs to be implemented (may be it is possible to support differently, this was just the most obvious idea).

**Comment by vifreefly** [comment]
>  It means all `PartOf` services get `JOB_RELOAD_OR_START` which is then collapsed into simple `JOB_RELOAD` for any active unit.

The interesting thing is, if no one child service has option `ReloadPropagatedFrom` (or the main target don't have `PropagatesReloadTo`), then command `reload-or-restart main.target` works for all child services, and all of them gets restarted (restarted, not reloaded).

**Comment by arvidjaar** [comment]
> The interesting thing is, if no one child service has option `ReloadPropagatedFrom` (or the main target don't have `PropagatesReloadTo`), then command `reload-or-restart main.target` works for all child services, and all of them gets restarted (restarted, not reloaded).

Yes, in this case it is assumed that target unit cannot do reload and so request is mapped to `JOB_RESTART` which is then propagated to dependent units.
